pyispyb/core/routes/legacy/data_collections.py

```diff
@@ -20,18 +20,21 @@
 """
 
 from fastapi import Depends
+from pyispyb.app.base import AuthenticatedAPIRouter
 from pyispyb.core.modules.legacy import data_collections
 from pyispyb.core.routes.legacy.dependencies import session_authorisation
 
-from .base import router
+from .base import router as legacy_router
+router = AuthenticatedAPIRouter(prefix="/data_collections", tags=["Data collections - legacy with header token"])
 
 
 __license__ = "LGPLv3+"
 
 
-@router.get(
+@legacy_router.get(
     "/{token}/proposal/session/{session_id}/list",
 )
+@router.get("/groups/session/{session_id}")
```

Comment:
My idea was to offer a better way to consume them, without this dangerous token in path, so that if we need one of these existing endpoints for future developments, we can just use it like this immediately, and then transparently re-implement it in a better way by just moving the route declaration somewhere else.

Reply:
Ahh right, i would maybe prefix the router with `/legacy` then? Is that what `generate_unique_id_function` is needed for, clashing routes?